commit 5f97a2d3f8880eb68ce2e0fd44285b6adc38293e
Author: Hugo Pereira Da Costa <hugo.pereira@free.fr>
Date:   Mon Aug 25 07:34:35 2014 +0200

    properly handle GTK_STATE_FLAG_CHECKED when rendering check buttons or
    radio buttons
    CCBUG: 338538

diff --git a/src/oxygenstyleoptions.h b/src/oxygenstyleoptions.h
index 52334df..a2712f8 100644
--- a/src/oxygenstyleoptions.h
+++ b/src/oxygenstyleoptions.h
@@ -76,7 +76,7 @@ namespace Oxygen
             if( flags & GTK_STATE_FLAG_INSENSITIVE ) (*this) |= Disabled;
             if( flags & GTK_STATE_FLAG_PRELIGHT ) (*this) |= Hover;
             if( flags & GTK_STATE_FLAG_SELECTED ) (*this) |= (Selected|Active);
-            if( flags & GTK_STATE_FLAG_ACTIVE ) (*this) |= Sunken;
+            if( flags & ( GTK_STATE_FLAG_ACTIVE | GTK_STATE_FLAG_CHECKED ) ) (*this) |= Sunken;
 
             // TODO: check whether one should use this, or gtk_widget_has_focus
             if( flags & GTK_STATE_FLAG_FOCUSED ) (*this) |= Focus;
@@ -90,7 +90,7 @@ namespace Oxygen
             if( flags & GTK_STATE_FLAG_INSENSITIVE ) (*this) |= Disabled;
             if( flags & GTK_STATE_FLAG_PRELIGHT ) (*this) |= Hover;
             if( flags & GTK_STATE_FLAG_SELECTED ) (*this) |= (Selected|Active);
-            if( flags & GTK_STATE_FLAG_ACTIVE ) (*this) |= Sunken;
+            if( flags & ( GTK_STATE_FLAG_ACTIVE | GTK_STATE_FLAG_CHECKED ) ) (*this) |= Sunken;
             if( flags & GTK_STATE_FLAG_FOCUSED ) (*this) |= Focus;
             else if( GTK_IS_WIDGET( widget ) && gtk_widget_has_focus(widget) ) (*this)|=Focus;
         }
diff --git a/src/oxygenthemingengine.cpp b/src/oxygenthemingengine.cpp
index 0e6382e..39d02d2 100644
--- a/src/oxygenthemingengine.cpp
+++ b/src/oxygenthemingengine.cpp
@@ -1868,7 +1868,7 @@ namespace Oxygen
             // shadow type defines checkmark presence and type
             GtkShadowType shadow( GTK_SHADOW_OUT );
             if( state&GTK_STATE_FLAG_INCONSISTENT ) shadow = GTK_SHADOW_ETCHED_IN;
-            else if( state&GTK_STATE_FLAG_ACTIVE ) shadow = GTK_SHADOW_IN;
+            else if( state&(GTK_STATE_FLAG_ACTIVE|GTK_STATE_FLAG_CHECKED) ) shadow = GTK_SHADOW_IN;
 
             Style::instance().renderCheckBox( widget, context, x, y, w, h, shadow, options, data );
 
@@ -1955,7 +1955,7 @@ namespace Oxygen
             // shadow type defines checkmark presence and type
             GtkShadowType shadow( GTK_SHADOW_OUT );
             if( state&GTK_STATE_FLAG_INCONSISTENT ) shadow = GTK_SHADOW_ETCHED_IN;
-            else if( state&GTK_STATE_FLAG_ACTIVE ) shadow = GTK_SHADOW_IN;
+            else if( state&(GTK_STATE_FLAG_ACTIVE|GTK_STATE_FLAG_CHECKED) ) shadow = GTK_SHADOW_IN;
             Style::instance().renderRadioButton( widget, context, x, y, w, h, shadow, options, data );
 
         } else {
